dm/docs/RFCS/20220202_optimist_shard_mode_support_conflict_ddl.md

```diff
@@ -0,0 +1,155 @@
+- joined: Join(final_tables), i.e. the set of all final_tables, i.e. the downstream joined table structure
+
+#### Algorithm
+
+We implement the final coordination algorithm according to the above definition.
+
+##### Initialization
+
+```golang
+// At the beginning, prev_tables is equal to the table structure of all sharding tables in the checkpoint
+prev_tables = {prev_tb1, prev_tb2,... ,prev_tbN}
+post_tables = {}
+final_tables = {prev_tb1, prev_tb2,... ,prev_tbN} = prev_tables
+```
+
+##### Maintenance
+
+When master receives a DDL for sharding table `X`, `compare(prev_tbx,post_tbx)` to find out if the DDL is a conflicting DDL, e.g. 
+
+- alter table add column new_col1, Compare(prev_tbx,post_tbx) = 1
+- alter table rename a to b, Compare(prev_tbx,post_tbx) = error
+- alter table change a from int to varchar, Compare(prev_tbx,post_tbx) = error
+- alter table add column a int not null, Compare(prev_tbx,post_tbx) = error
+
+If the DDL is not a conflicting DDL, just update the `tbx` in `prev_tables` and coordinate it normally.  
+If the DDL is a conflicting DDL, add `post_tbx` to `post_tables`, update `final_tables`, i.e.
+
+```golang
+// Receive conflict DDL for sharding table X, get prev_tbx, post_tbx
+prev_tables = {prev_tb1,... ,prev_tbx,... ,prev_tbN}
+post_tables = {post_tbx}
+final_tables = {prev_tb1,... ,post_tbx,... ,prev_tbN}
+```
+
+##### Termination
+
+master will then receive conflicting DDLs for the different sharidng tables, keeping `post_tables` and `final_tables` updated.
+
+```golang
+// M <= N
+prev_tables = {prev_tb1, prev_tb2,...,prev_tbN}
+post_tables = {post_tb1,post_tb3,...,post_tbM}
+final_tables = {post_tb1, prev_tb2, post_tb3,...,post_tbM,...,prev_tbN}
+joined=Join(final_tables)
+```
+
+Coordination of all sharding tables is completed when the following condition is finally reached.
+
+- `joined ! = error`
+- `Compare(joined,prev_tbx) == error` for any sharding table `X` in `post_tables`
+- for any sharding table `X` in `post_tables`, for any sharding table `Y` in `final_tables`: `Compare(Join(prev_tbx,final_tabley),post_tbx) >= 0`
+
+##### Proof
+
+We just need to prove that the final state of Termination satisfies the goal we proposed earlier, i.e., for each conflicting DDL(-a,+b), Termination's final end condition satisfies: all sharding tables (final_tables) have no `a` but `b`
+
+Assume that sharding table `K` executes a DDL1 such that tbk(-a,+b), i.e., *a ∈ prev_tbk*, *b ∉ prev_tbk*, *a ∉ post_tbk*, *b ∈ post_tbk*, and *post_tbk = prev_tbk - {a} + {b}* ①
+
+1. joined ! = error  
+
+    i.e. all the sharding tables (final_tables) have no conflicts, substituting *table=k* gives *(joined=Join(final_tables)=Join(final_tbk,...) =Join(post_tbk,...)*  
+    ⟶ *Compare(joined,post_tbk)=Compare(Join(post_tbk,...) ,post_tbk)>=0*  
```

Comment:
```suggestion
    ⟶ *Compare(joined,post_tbk)=Compare(Join(post_tbk,...),post_tbk)>=0*  
```

why we jump to "Compare(joined,post_tbk)"? Do you mean "considering calculating the bool Compare(joined,post_tbk) != error, it equals to Compare(Join(post_tbk,...),post_tbk)>=0"?